### DISCUSSION TITLE
Using rotary axes

### DISCUSSION THREAD

**Discussion by macbef** [grblHAL/core#301] [General]
Hello,
I have a basic question:

If I configure grbl with 4 linear axes (XYZA) and one rotary axis (B), how do I specify the radius of my workpiece that is mounted on B?

How does grbl know which linear axes are affected by the radius?

Think of this as a lathe with 2 sledges. X and Z are moving in B's radial direction. Y and A are parallel to the B-axis.

So my first question is: If I'm only using X,Y and B and I want to mill on the surface of a cylinder, X can be constant. When grbl calculates the velocity of the path, it has to be aware of the radius. How do I define this radius? Is this the tool length offset?

And second question: Can I define two different tool length offsets for the same machine?

Thanks!!

**Comment by terjeio** [comment]
>  how do I specify the radius of my workpiece that is mounted on B?

In the CAM software. [ioSender](https://github.com/terjeio/ioSender) will also need radius input when wrapping X or Y to a rotary axis.

> Can I define two different tool length offsets for the same machine?

I am not sure I understand. It is possible to compile grblHAL with a [tool table](http://svn.io-engineering.com/grblHAL/html/config_8h.html#aef2982cbc62fbf11dd7bb9bd3f5f976e) that keeps the tool offsets in non-volatile storage. And it is possible to specify tool offsets for all axes unless a compatibility mode limits it to one as for legacy Grbl. And tool offsets can be changed programmatically at all times. See [G10 L1, G10 L10, G10 L12](http://linuxcnc.org/docs/2.6/html/gcode/gcode.html#sec:G10-L1_) and [G43.x](http://linuxcnc.org/docs/2.6/html/gcode/gcode.html#sec:G43). Note the `R I J Q` words are not supported by grblHAL.

**Reply by macbef** [reply]
So if B is my rotary axis, can I use G10 L1 P1 B200 to set the radius to 200mm?
(Actually, if I try I get error:20)

**Reply by terjeio** [reply]
> So if B is my rotary axis, can I use G10 L1 P1 B200 to set the radius to 200mm?

No, the diameter is needed by the CAM software when generating the gcode (it might be taken from the CAD drawing).
If your CAM software do not support rotary axes you can try the ioSender wrap functionality, or if the sender you use supports wrapping use that.

FYI you may run into a feedrate issue with rotary axes - disussed [here](https://github.com/grblHAL/core/discussions/241#discussioncomment-4718602).

> (Actually, if I try I get error:20)

This is due to the tool table not beeing compiled in, error 20 is unsupported gcode.
